wporg-two-factor.php

```diff
@@ -35,9 +35,14 @@ function two_factor_providers( array $providers ) : array {
 	$desired_providers = array(
 		'Two_Factor_WebAuthn'     => '',
 		'Two_Factor_Totp'         => '',
-		'Two_Factor_Backup_Codes' => '',
```

Comment:
Will this cause any problems w/ the flow in #18? The provider [wouldn't be loaded](https://github.com/WordPress/two-factor/blob/555d88685087c0f9e08030cf93bdacfb749dbded/class-two-factor-core.php#L133-L151) when the user visits the page, so we might not be able to generate the backup codes during activation, unless we explicitly `require` it in the Ajax callback.

The `two_factor_enabled_providers_for_user` might solve any issues there, and seems a bit more appropriate.

Reply:
Possibly - although for the activation flow we could add in the backup codes through hooking in later to `two_factor_providers` since we'd be manipulating the standard flow anyway.

Reply:
🤔 yeah, fair point. what's problem is this trying to solve? maybe there's another way to do it?

Reply:
> what problem is this trying to solve?

In #21 we were trying to avoid users setting up backup codes without another preferred method, but maybe it's better just to prompt them to do so, rather than force them to using code?

The plugin does have logic that if you only have one provider it becomes the primary, so potentially in this scenario should there be an error in setup for a preferred provider the user could become unnecessarily locked out.

Reply:
Would the ["enabled" vs "available" distinction](https://github.com/WordPress/wporg-two-factor/pull/24#discussion_r1102046729) might be useful in avoiding locking the user out?